ansible_runner/config/_base.py

```diff
@@ -516,6 +530,21 @@ def wrap_args_for_containerization(self, args, execution_mode, cmdline_args):
         logger.debug(f"container engine invocation: {' '.join(new_args)}")
         return new_args
 
+    def _generate_container_auth_file(self, host, username, password):
+        token = "{}:{}".format(username, password)
+        encrypted_container_auth_data = {'auths': {host: {'auth': b64encode(token.encode('UTF-8')).decode('UTF-8')}}}
+        # Create a new temp file with container auth data
+        path = tempfile.mkdtemp(prefix='ansible_runner_registry_%s_' % self.ident)
```

Comment:
Should this file be in `self.private_data_dir` ?

I think this would make it easier to clean up & debug.

Reply:
no, it can't be there, that's the point

these are credentials for pulling the container image. We specifically intend to _not_ make those secrets available to the playbook. The playbook runs in the volume mounted private_data_dir. That rules out that folder, which is why we started doing this in a wrapper folder, but that doesn't get transmitted.

Reply:
Thanks @AlanCoding I see why this shouldn't be in `self.private_data_dir`.

I still have concerns about cleanup and debugging. I can't remember, is the parent directory of `self.private_data_dir` something we could use? 

I _do_ like that the `mkdtemp` call uses a prefix so that, given a job id, we can trace/find the directory that this `auth.json` lives.

Reply:
> I do like that the mkdtemp call uses a prefix so that, given a job id, we can trace/find the directory that this auth.json lives.

We plan on doing a last-ditch cleanup action in case something happens that prevents the `atexit` methods from running and deleting the directories. For that last-resort cleanup, we will do the same thing we did with isolated jobs. For this, we don't look for a specific job id, but instead list directories based on a pattern and exclude job ids which may still be running.

At the moment, however, we don't yet have cleanup for the `private_data_dir` on execution nodes _at all_. This was just never programmed into `ansible-runner worker` in the first place. There are no major issues in doing any of this, we just need to get it done. Our current plan still necessitates several additions to runner, and I just need to get a good cadence of merges happening.